### ISSUE TITLE
Viewer bottom view does not load slides properly.

### ISSUE THREAD

**Issue by mohitKS777** [openseadragon/openseadragon#2309]
I am rendering WSI images from a S3 bucket. In some cases we are getting images properly in the bottom view, in some cases we are not. Is this a known issue in openseadragon. Adding a screenshot for further clarification.
![Screenshot (13)](https://user-images.githubusercontent.com/98010762/222076255-4c0ac618-03a7-426c-9ef4-c0df8054351f.png)

**Comment by msalsbery** [comment]
@mohitKS777 The navigator image? When it doesn't show, is there failed image tile requests?

**Comment by mohitKS777** [comment]
Yes, I also get an error in the console. But it is only a few selected slides.

**Comment by iangilman** [comment]
What error are you getting? What format are these images in?

**Comment by mohitKS777** [comment]
I am using simple svs that gets converted to dzi in my aws lambda server. Regarding the error I get "Cannot load tile {url}".

**Comment by iangilman** [comment]
Sounds like maybe your DZI converter doesn't make the smaller levels? If you look in the "_files" folder of the resulting DZI, does it have folders going all the way down to 0?

**Comment by mohitKS777** [comment]
I can't check this , as it is getting stored in a S3 bucket which my organization owns (I don't have access to it). I am using libvips in my lambda server for converting slides. So I am assuming it generates smaller levels.

**Comment by iangilman** [comment]
Hmm. Can you look in the browser inspector in the network tab and see what files are getting errors? It may be clear from that whether it's the "tip of the pyramid". You're right, though, that libvips should be generating the entire pyramid. At any rate, it would be useful to know which tiles aren't getting loaded.

**Comment by K3TH3R** [comment]
I'm seeing something similar in our OSD implementation, but it appears to only be in v4. If I downgrade to the 3.1.0 release without changing *any* code, the navigator starts working again.

**Comment by Aiosa** [comment]
I tested the navigator with detached and container modes with different sizes and crazy aspect ratios. While AR of 1:8 seems to do some wild things, normal ARs seemed fine in all cases with v4+ DZI-based protocols. I am afraid we won't be able to help with such little information though...

**Comment by iangilman** [comment]
Yeah, it would be lovely to have a repro... @mohitKS777 @K3TH3R Can either of you share an image that fails in the navigator?

**Comment by K3TH3R** [comment]
@Aiosa @iangilman I can't speak to issues with DZI tiles, but as you can see in the below screenshots, by changing nothing other than OSD 3.1.0 to 4.0, the navigator stops functioning the same. You can see the code and try swapping the dependencies yourself on the CodePen here: https://codesandbox.io/s/openseadragon-4-0-navigator-issue-4orrhp?file=/src/OSDWrapper.js

![Screenshot 2023-05-01 at 9 57 17 am](https://user-images.githubusercontent.com/11901111/235382115-701e3cca-cc04-4508-ba27-89c878c16bd5.png)

![osd-4](https://user-images.githubusercontent.com/11901111/235382120-5cc0583e-ea06-4bc6-b378-d420aafe3584.png)

**Comment by Aiosa** [comment]
In your example, things indeed break. It breaks both for IIIF and simple tilesource and since all is well with the set-up we are using I guess the difference could be between using browser VS node execution environment.

I do not have time unfortunately now to try to fix this, but I've managed to narrow down the issue:

in the navigator, its tile size is computed in units of pixels since
`_positionTile` calls `viewport.deltaPixelsFromPointsNoRotate` and while `viewer.viewport` has `getZoom = 1`, `viewer.navigator.viewport` has `getZoom = 0.00454253472222222` which seems inconsistent.

Outside the navigator
![image](https://user-images.githubusercontent.com/34658867/235419110-a66f08a4-7012-477b-8c0a-88d1b8bd9213.png)
Inside the navigator
![image](https://user-images.githubusercontent.com/34658867/235419151-29122f74-3775-4f80-88cf-31d0969777e6.png)

These values go directly to the canvas drawing so the positioning logics then works inside the navigator with [1px, 1.2px] large tiles.

**Comment by iangilman** [comment]
I suspect this has to do with https://github.com/openseadragon/openseadragon/pull/2256 by @pearcetm. 

As far as I can tell, here's what's happening: The navigator is set up to be absolutely positioned, and the width is given but no height. This creates, for a brief moment, a wide but very short navigator, so the zoom is set to something very tiny so that an image would still be viewable in its entirety. Then, after the image has been opened, the height is set. This doesn't change the zoom, but does stretch the tiny image vertically. 

Here's a workaround: In the options, set the `navigatorHeight` to something large, like `'200px'`. Then, after the viewer is created, add this line: 

```
this.#viewer.navigator.autoResize = true;
```

The `navigatorHeight` change means we never get to that super short state where the zoom goes way out. The `autoResize` change means when the `setHeight` is called, we properly change the zoom, rather than stretching the image. 

@pearcetm What do you think? #2256 changed the `autoResize` to always be false, but maybe this is a case where it shouldn't be? 

If you make the `autoResize` change, but not the `navigatorHeight`, we don't get the stretching anymore, but the image does still stay small. That may just be how it is with the new behavior for the autoResize. 

BTW, part of the reason @mohitKS777 is doing the `setHeight` is because `navigatorSizeRatio` doesn't work if `navigatorPosition` is `'ABSOLUTE'`. I wonder if there's a good reason or if that could be fixed?

**Comment by pearcetm** [comment]
For some reason I'm getting a CORS error with your image in the codepen example, so it's hard to test.

> @pearcetm What do you think? #2256 changed the `autoResize` to always be false, but maybe this is a case where it shouldn't be?

Could be. I haven't worked through all the calculations/events that are going on yet. I suspect it isn't the most efficient fix though, even if it might work, because in most (though apparently not all) scenarios, `Navigator.update` triggers resize logic that overrides the default implementation. However, when `_resizeWithViewer == false` like it is in this case, the navigator's update->resize is bypassed.

I wonder if the navigator should listen to its own `open` event and call `this.updateSize()` which should theoretically take care of all calculations. `updateSize()` should probably also be called from `setWidth()` and `setHeight()`, both of which disable `_resizeWithViewer`.

**Comment by iangilman** [comment]
Yeah, I got the CORS error too... Weird that JPEG is doing that? Maybe because we're putting it in a canvas? Anyway, I just replaced it with the duomo DZI. 

Excellent point about `updateSize`! We should just add that to `setWidth` and `setHeight` like you say. That's probably the complete fix for this issue! 

@mohitKS777 That's also a better workaround than the one I gave you. Just call `this.#viewer.navigator.updateSize();` after the `setHeight` call. 

Would either of you be up for making a patch?

**Comment by Aiosa** [comment]
I got cors error for all except chrome browser (only non-incognito mode). Which is weird since I also checked that and all seemed well set, even according to the mdn docs.

**Comment by pearcetm** [comment]
I've created a PR that hopefully addresses the issue. I'm struggling to get the repro code to work (when I replace the image with CORS problems with other JPEGs the navigator works fine for me; the duomo dzi just doesn't open at all for some reason). It'd be great if someone who can reproduce the issue can give the updated code a try to see if it resolves the problem.

**Comment by iangilman** [comment]
@pearcetm I have confirmed that the fix works. Thank you for making the PR!
